### PR TITLE
CONFIGURE: Deduplicate simple compiler flag support tests

### PR DESCRIPTION
**Pull request by orgads**

### CODE REVIEW

**Comment by orgads** [comment]
I see that @aquadran fixed the `packed-member` warning in GLK recently. So should we leave it enabled?

File: configure [hunk 1]
```diff
@@ -2297,6 +2297,8 @@ set_flag_if_supported "-Wglobal-constructors"
 # unknown -Wno-xxx flags.
 set_flag_if_supported -Wno-pragma-pack
 
+set_flag_if_supported -Wno-address-of-packed-member
+
```
Comment (aquadran) [inline configure:2301]: I think it's better to fix code, than silence compiler globally. we care about alignment. compiler warns for some reason. @dreammaster any idea about code?

Comment (orgads) [reply configure]: Removed it for now.

File: configure [hunk 2]
```diff
@@ -2313,11 +2295,7 @@ echo $_no_undefined_var_template
 # macros.
 # Note: as above, we check for -Wpragma-pack as not all compilers error out on
 # unknown -Wno-xxx flags.
-echocheck "whether C++ compiler accepts -Wno-pragma-pack"
-echo "int main() { return 0; }" >$TMPC
-cc_check -Wpragma-pack -Werror && _no_pragma_pack=yes
-test "$_no_pragma_pack" = yes && append_var CXXFLAGS "-Wno-pragma-pack"
-echo $_no_pragma_pack
+set_flag_if_supported -Wno-pragma-pack
```
Comment (lephilousophe) [inline configure:2298]: This is not the same as what have been done before.
Before, we checked support of the positive flag and enabled the negative one.

File: configure [hunk 3]
```diff
@@ -2275,35 +2272,20 @@ if test "$_release_build" = yes; then
...
+		echo yes
+	else
+		echo no
+	fi
+}
 
-if test "$_no_undefined_var_template" = yes; then
-	append_var CXXFLAGS "-Wno-undefined-var-template"
-fi
-echo $_no_undefined_var_template
```
Comment (lephilousophe) [inline configure:2306]: This whole check shouldn't be removed.
The PR got reverted in 940b2a20f1cd490afb6e541a3cd26f0d3bdd1687.

File: devtools/create_project/xcode.cpp [hunk 4]
```diff
@@ -1081,7 +1081,6 @@ void XcodeProvider::setupBuildConfiguration(const BuildSetup &setup) {
 	ADD_SETTING(scummvm_Debug, "GCC_WARN_UNUSED_FUNCTION", "YES");
 	ValueList scummvm_WarningCFlags;
 	scummvm_WarningCFlags.push_back("-Wno-multichar");
-	scummvm_WarningCFlags.push_back("-Wno-undefined-var-template");
```
Comment (lephilousophe) [inline devtools/create_project/xcode.cpp:1084]: Same as above, this shouldn't be removed.

**Review by lephilousophe — COMMENTED** [review]
I made some comments inline.

File: configure (hunk 2 above)
Comment (orgads) [reply configure:2298]: Does it matter?

Comment (orgads) [reply configure:2298]: Ok, I see it does :)

I'll try to find a better way.

File: configure (hunk 3 above)
Comment (orgads) [reply configure:2306]: Which compilers have this warning? I don't have it with MinGW/GCC12.

Comment (orgads) [reply configure:2306]: Reverted.

Comment (lephilousophe) [reply configure:2306]: Maybe clang?

Comment (orgads) [reply configure:2306]: Yes. I see it here: https://github.com/orgads/scummvm/runs/6715783766?check_suite_focus=true#step:8:474

Reverted anyway, thank you.

File: configure [hunk 5]
```diff
@@ -2275,49 +2272,38 @@ if test "$_release_build" = yes; then
...
+	else
+		echo no
+	fi
+}
 
-if test "$_global_constructors" = yes; then
-	append_var CXXFLAGS "-Wglobal-constructors"
-fi
-echo $_global_constructors
+set_flag_if_supported "-Wglobal-constructors"
```
Comment (bluegr) [inline configure:2290]: A bit pedantic, but here the argument is enclosed in quotes, but it isn't enclosed in the other cases. Let's decide on a style (with or without quotes) and use it across all cases

**Comment by bluegr** [comment]
Nice work! @orgads, please address the issue above in order to merge this

File: configure (hunk 5 above)
Comment (orgads) [reply configure:2290]: Done

**Comment by bluegr** [comment]
Thanks for your work! Merging

**Pull request merged**